src/main/java/com/softserve/academy/spaced/repetition/controller/UserController.java

```diff
@@ -34,12 +36,15 @@
     @Autowired
     PointsTransactionService pointsTransactionService;
 
+    @Autowired
+    private ApplicationEventPublisher publisher;
+
     @GetMapping("api/user/details")
     @PreAuthorize("isAuthenticated()")
     public ResponseEntity<UserDTO> getAuthorizedUserPublicInfo() throws NotAuthorisedUserException {
         User user = userService.getAuthorizedUser();
         if (!userService.isAdmin(user)) {
-            userService.updatePointsBalance(user);
+            publisher.publishEvent(new PointsBalanceEvent(this, user));
```

Comment:
Please do not pass `this`. `this` is a managed Spring bean with different dependencies and access to different methods, that an event shouldn't access.